dspy/predict/predict.py

```diff
@@ -20,9 +21,19 @@ def reset(self):
         self.demos = []
 
     def dump_state(self):
-        state_keys = ["lm", "traces", "train", "demos"]
+        state_keys = ["lm", "traces", "train"]
         state = {k: getattr(self, k) for k in state_keys}
 
+        state["demos"] = []
+        for demo in self.demos:
+            demo = demo.copy()
+
+            for field in demo:
+                if isinstance(demo[field], BaseModel):
+                    demo[field] = demo[field].model_dump_json()
```

Comment:
That's the thing, I'm assuming I don't have to, because the demos of the predictors don't need to keep the same types after saving and loading their states, at least after this commit https://github.com/stanfordnlp/dspy/commit/b9516c45dbcdedbed6b71dab7cd4a6c1024e0dcf .

When the state is loaded, typed demos are converted to a dict of str fields, which are injected correctly when the pipeline is runned.